modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -3361,15 +3361,15 @@ protected List<MappedModel> getAllOfDescendants(String thisSchemaName, OpenAPI o
                 break;
             }
             currentSchemaName = queue.remove(0);
-            MappedModel mm = new MappedModel(currentSchemaName, toModelName(currentSchemaName));
-            descendentSchemas.add(mm);
             Schema cs = schemas.get(currentSchemaName);
             Map<String, Object> vendorExtensions = cs.getExtensions();
-            if (vendorExtensions != null && !vendorExtensions.isEmpty() && vendorExtensions.containsKey("x-discriminator-value")) {
-                String xDiscriminatorValue = (String) vendorExtensions.get("x-discriminator-value");
-                mm = new MappedModel(xDiscriminatorValue, toModelName(currentSchemaName));
-                descendentSchemas.add(mm);
-            }
+            String mappingName =
+                Optional.ofNullable(vendorExtensions)
+                    .map(ve -> ve.get("x-discriminator-value"))
+                    .map(discriminatorValue -> (String) discriminatorValue)
+                    .orElse(currentSchemaName);
```

Comment:
This logic is changing - since this value previously would have been `null`

Reply:
No, prior to this change `currentSchemaName` was added as a mapping name on line 3365, and then another mapping was added on line 3371 if an `x-discriminator-value` was set. With this change, it's one or the other, not both.

Reply:
I wonder if this change is causing the issues you're seeing with Dart. I'd have to look more closely at how the schemas are being used later on, but this adds _less_ schemas now; so perhaps some filterins logic is applied at some point?

I would try reverting this part of the change and see what the Dart output looks like.